test/util.test.js

```diff
@@ -468,4 +470,375 @@ describe('mergeOptions', function () {
   });
 
 });  // mergeOptions
-});  // util
+
+  describe('recursiveDOMDelete', function () {
+    beforeEach(function() {
+      this.jsdom_global = jsdom_global();
+    });
+
+    afterEach(function() {
+      this.jsdom_global();
+    });
+
+    it('removes children', function () {
+      var root = document.createElement("div");
+      // Create children for root
+      var parent = document.createElement("div");
+      var parentSibiling = document.createElement("div");
+      // Attach parents to root
+      root.appendChild(parent);
+      root.appendChild(parentSibiling);
+      // Create children for the respective parents
+      var child = document.createElement("div");
+      var childSibling = document.createElement("div");
+      // Attach children to parents
+      parent.appendChild(child);
+      parentSibiling.appendChild(childSibling);
+
+      util.recursiveDOMDelete(root);
+      assert.equal(root.children.length, 0);
+      assert.equal(parent.children.length, 0);
+      assert.equal(parentSibiling.children.length, 0);
+      assert.equal(child.children.length, 0);
+      assert.equal(childSibling.children.length, 0);
+    });
+  });
+
+  describe('isDate', function () {
```

Comment:
Expecting a test on bad values here, to eliminate false positives.
With garbage input, near dates (e.g. 1 past en dof month, leap years feb)